include/mbedtls/build_info.h

```diff
@@ -77,7 +77,11 @@
 #if defined(MBEDTLS_PK_C) && defined(MBEDTLS_USE_PSA_CRYPTO)
 #define MBEDTLS_PK_WRITE_C
 #endif
-#if defined(MBEDTLS_PSA_CRYPTO_CONFIG)
+
+/* Make sure all configuration symbols are set before including check_config.h,
+ * even the ones that are calculated programmatically. */
+#if defined(MBEDTLS_PSA_CRYPTO_CONFIG) /* PSA_WANT_xxx influences MBEDTLS_xxx */ || \
+    defined(MBEDTLS_PSA_CRYPTO_C) /* MBEDTLS_xxx influences PSA_WANT_xxx */
```

Comment:
I don't think this is right. The only defines that are outside of `MBEDTLS_PSA_CRYPTO_CONFIG` are:
```
#define PSA_WANT_KEY_TYPE_DERIVE 1
#define PSA_WANT_KEY_TYPE_RAW_DATA 1
```
And there are no `PSA_WANT` defines in `check_config.h`. 
I can see that this kind of validation is done in `check_crypto_config.h`. Can we have the checks that use `PSA_WANT_XXX`, added in this PR - there?

Reply:
This is the result of the discussion that can be found in the issue. The topic started here: https://github.com/Mbed-TLS/mbedtls/issues/6132#issuecomment-1235274278